htdocs/product/admin/product.php

```diff
@@ -653,12 +653,11 @@
 }
 print '</tr>';
 
-if (empty($conf->global->PRODUIT_USE_SEARCH_TO_SELECT)) {
-	print '<tr class="oddeven">';
-	print '<td>'.$langs->trans("NumberOfProductShowInSelect").'</td>';
-	print '<td class="right"><input size="3" type="text" class="flat" name="value_PRODUIT_LIMIT_SIZE" value="'.$conf->global->PRODUIT_LIMIT_SIZE.'"></td>';
-	print '</tr>';
-}
+// Limit of search results
+print '<tr class="oddeven">';
+print '<td>'.$langs->trans("NumberOfProductShowInSelect").'</td>';
+print '<td class="right"><input size="3" type="text" class="flat" name="value_PRODUIT_LIMIT_SIZE" value="'.$conf->global->PRODUIT_LIMIT_SIZE.'"></td>';
```

Comment:
The goal of option PRODUIT_USE_SEARCH_TO_SELECT is to have limited by the first char we entered instead of a setup limit.
If you make the option PRODUIT_LIMIT_SIZE working in both case (when PRODUIT_USE_SEARCH_TO_SELECT is on and off), you must also check that the default value for limit when PRODUIT_USE_SEARCH_TO_SELECT is on, remains to value "0" (no limit). So we keep the expected behaviour:
"if we choosed to have a limit managed by the char entered instead of a setup, we must not have by default a list restricted".

So into conf.class.php we should change
```
if (!isset($this->global->PRODUIT_LIMIT_SIZE)) {
				$this->global->PRODUIT_LIMIT_SIZE = 1000;
}

```
into
```
if (!isset($this->global->PRODUIT_LIMIT_SIZE)) {
  if (empty($conf->global->PRODUIT_USE_SEARCH_TO_SELECT)) {
				  $this->global->PRODUIT_LIMIT_SIZE = 1000;
  } else {
				  $this->global->PRODUIT_LIMIT_SIZE = 0;
  }
}
```

So, with a default setup, when we never set any limit value, if we choose "limit by entered keys", the combo list will be complete and not truncated (we experience a lot of issue of end users asking why the list is not "complete" so it must not be truncated by default when using ajax mode).